src/Microsoft.Diagnostics.EventFlow.Core/Implementations/EventData.cs

```diff
@@ -28,6 +29,8 @@ public class EventData: IDeepCloneable<EventData>
 
         public long Keywords { get; set; }
 
+        public object EventContext { get; set; } = ContextResolver.GetRequestContext<object>();
```

Comment:
I wonder if we should put this behind a flag. That is, by default, correlation functionality is turned off. If user turns it on in the config file, or through some switch on DiagnosticsPipeline, then we switch it on. The reason is that correlation usage has some, eventhough it's not significant, performance overhead.